Source/AST/Nodes/Node.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  Node.swift
+//  Down
+//
+//  Created by John Nguyen on 07.04.19.
+//
+
+import Foundation
+import libcmark
+
+public typealias CMarkNode = UnsafeMutablePointer<cmark_node>
+
+/// A node is a wrapper of a raw `CMarkNode` belonging to the abstract syntax tree
+/// generated by cmark.
+///
+public protocol Node {
+    /// The wrapped node.
+    var cmarkNode: CMarkNode { get }
+    
+    /// The wrapped child nodes.
+    var childen: [Node] { get }
+}
+
+extension Node {
+    /// The children of this node.
+    public var childen: [Node] {
+        var result: [Node] = []
+        var child = cmark_node_first_child(cmarkNode)
+        
+        while let raw = child {
+            
+            guard let node = raw.wrap() else {
+                assertionFailure("Couldn't wrap node of type: \(raw.type)")
+                continue
+            }
+            
+            result.append(node)
+            child = cmark_node_next(child)
+        }
+        
+        return result
+    }
+    
+    /// True iff the node has a sibling that succeeds it.
+    public var hasSuccessor: Bool {
+        return cmark_node_next(cmarkNode) != nil
+    }
+}
+
+
+// MARK: - Helper extensions
+
+public extension UnsafeMutablePointer where Pointee == cmark_node {
+    
+    /// Wraps the cmark node referred to by this pointer.
+    public func wrap() -> Node? {
```

Comment:
Nit: `public` is redundant here w/  so Xcode will show a warning

Reply:
A right, I was in Xcode 10.1 and it doesn't show the warning there. I'll update it.